src/ui/components/ShowMoreCard/index.js

```diff
@@ -29,15 +28,13 @@ export class ShowMoreCardBase extends React.Component {
   }
 
   componentWillReceiveProps() {
-    this.truncateToMaxHeight(ReactDOM.findDOMNode(this.contents));
+    this.setState({ expanded: true }, () => {
+      this.truncateToMaxHeight(this.contents);
```

Comment:
This will work in production but I don't think it will work for our test suite. The problem is that React [does not guarantee](https://facebook.github.io/react/docs/react-component.html#setstate) that the callback to `setState()` will be invoked immediately. Thus, the test you wrote will probably pass *most* of the time but not all of the time. If the `setState()` callback is delayed then your test will fail.

There are a few ways to solve this. We could try to wedge promises in here so that the test can use a promise chain. I don't see an easy way to do that.

Alternatively, we could keep the process synchronous which would make the test you already wrote pass 100% of the time. I think restoring `componentWillReceiveProps()` to how it was and adding this `else` block would accomplish that:

```js
truncateToMaxHeight = (contents) => {
  // If the contents are short enough they don't need a "show more" link; the
  // contents are expanded by default.
  if (contents.clientHeight > MAX_HEIGHT) {
    this.setState({ expanded: false });
  } else {
    this.setState({ expanded: true });
  }
}
```

Reply:
Well, it's possible that Enzyme has some trickery to ensure all batched state changes are applied but they don't document this. I'm skeptical since this kind of state change is triggered from *within* the component (not from an Enzyme method).

Reply:
What we could do is reuse the `callback` from Enzyme `setState()` (see: https://github.com/airbnb/enzyme/pull/617) and ensure the test case asserts things once the `setState()` has been triggered (and use `done()` to ensure the test does not silently fail)

Reply:
Ugh, yeah, we really can't rely on `setState()` being synchronous anywhere. https://github.com/airbnb/enzyme/issues/509#issuecomment-237362893

It would be nice to use promises instead of `done()` but your approach will work. Creating a promise wrapper for everywhere we rely on `wrapper.setState()` in the tests is probably a better long term solution.